src/Common/Common.Api.Test/GatewayClientTests.cs

```diff
@@ -498,5 +499,231 @@ public async Task SetOnPremisesDataGatewayTenantPolicyCanBeSerialized()
+        {
+            // Arrange
+            var client = Utilities.GetTestClient("");
+            var request = new UserAccessRightEntry()
+            {
+                DatasourceUserAccessRight = DatasourceUserAccessRight.Read,
+                DisplayName = "the user displayName",
+                Identifier = Guid.NewGuid().ToString(),
+                PrincipalType = PrincipalType.User,
+                UserEmailAddress = "theEmailAddress@foo.com",
+            };
+
+            // Act
+            var result = await client.AddUsersToGatewayClusterDatasource(Guid.NewGuid(), Guid.NewGuid(), request, true);
+
+            // Assert
+            result.IsSuccessStatusCode.Should().Be(true);
+        }
+
+        [TestMethod]
+        public async Task GetOnPremisesDataGatewayClusterDatasourceUsersCanBeSerializedAndDeSerialized()
+        {
+            // Arrange
+            var userEmailAddress = "theemailAddress@foo.com";
+            var serializedODataRepsonse = $@"{{
+    ""@odata.context"": ""http://example.net/v2.0/myorg/me/$metadata#users"",
+    ""value"": [
+        {{
+            ""emailAddress"":""{userEmailAddress}"",
+            ""datasourceAccessRight"":""Read"",
+            ""displayName"":""the email address"",
+            ""identifier"":""{userEmailAddress}"",
+            ""principalType"":""User""
+        }}
+    ]
+}}";
+            var client = Utilities.GetTestClient(serializedODataRepsonse);
+
+            // Act
+            var result = await client.GetGatewayClusterDatasourceUsers(Guid.NewGuid(), Guid.NewGuid(), true);
+
+            // Assert
+            var resultUsers = result.ToArray()[0];
+
+            // Assert
+            resultUsers.UserEmailAddress.Should().Be(userEmailAddress);
+        }
+
+        [TestMethod]
+        public async Task RemoveOnPremisesDataGatewayClusterDatasourceUserCanBeSerialized()
+        {
+            // Arrange
+            var client = Utilities.GetTestClient("");
+
+            // Act
+            var result = await client.RemoveGatewayClusterDatasourceUser(Guid.NewGuid(), Guid.NewGuid(), Guid.NewGuid().ToString(), true);
+
+            // Assert
+            result.IsSuccessStatusCode.Should().Be(true);
+        }
+
+        [TestMethod]
+        public async Task SetOnPremisesDataGatewayClusterDatasourceCredentialsCanBeSerializedAndDeSerialized()
+        {
+            // Arrange
+            var errorCode = Guid.NewGuid().ToString();
+            var serializedODataRepsonse = $@"{{
+    ""@odata.context"": ""http://example.net/v2.0/myorg/me/$metadata#credentials"",
+    ""value"": [
+        {{          
+            ""ErrorCode"":""{errorCode}"",
+            ""ErrorMessage"":""the error message"",
+            ""GatewayName"":""the gateway name"",
+        }}
+    ]
+}}";
+            var client = Utilities.GetTestClient(serializedODataRepsonse);
+            var request = new DatasourceCredentialDetails
```

Comment:
use UpdateGatewayClusterDatasourceCredentialRequest instead of DatasourceCredentialDetails

Reply:
I would like to keep this the same as in the backend code.